src/Storages/tests/gtest_low_priority_background_processing_pool.cpp

```diff
@@ -1,14 +1,84 @@
 #include <gtest/gtest.h>
 #include <Interpreters/Context.h>
+#include <Common/Config/ConfigProcessor.h>
 #include <Common/tests/gtest_global_context.h>
 #include <Storages/MergeTree/BackgroundProcessingPool.h>
+#include <iostream>
+#include <Poco/Util/Application.h>
+#include <daemon/BaseDaemon.h>
 
+DB::BackgroundProcessingPoolTaskResult odd(int *data) {
+    for (int i = 0; i < 10; ++i) {
+        if (i % 2) {
+            data[i] = i;
+        }
+    }
+    return DB::BackgroundProcessingPoolTaskResult::SUCCESS;
+}
 
+DB::BackgroundProcessingPoolTaskResult even(int *data) {
+    for (int i = 0; i < 10; ++i) {
+        if (!(i % 2)) {
+            data[i] = i;
+        }
+    }
+    return DB::BackgroundProcessingPoolTaskResult::SUCCESS;
+}
+
+static const char * minimal_default_user_xml =
+"<yandex>"
+"    <profiles>"
+"        <default></default>"
+"    </profiles>"
+"    <users>"
+"        <default>"
+"            <password></password>"
+"            <networks>"
+"                <ip>::/0</ip>"
+"            </networks>"
+"            <profile>default</profile>"
+"            <quota>default</quota>"
+"        </default>"
+"    </users>"
+"    <quotas>"
+"        <default></default>"
+"    </quotas>"
+"</yandex>";
+
+static DB::ConfigurationPtr getConfigurationFromXMLString(const char * xml_data)
+{
+    std::stringstream ss{std::string{xml_data}};
+    Poco::XML::InputSource input_source{ss};
+    return {new Poco::Util::XMLConfiguration{&input_source}};
+}
 
 TEST(LowPriorityBackgroundProcessingPool, SimpleCase) {
     using namespace DB;
     const auto & context_holder = getContext();
     Context ctx = context_holder.context;
 
+    ConfigurationPtr config = getConfigurationFromXMLString(minimal_default_user_xml);
+    ctx.setConfig(config);
+
+    int data[10];
```

Comment:
Please, try to avoid using raw pod arrays. Use std::vector instead, which will make it easier to manage the contents (find out elements count when making assignments)